src/Components/ImportMap.js

```diff
@@ -11,15 +11,20 @@ const ImportMap = () => {
 
   const { theme } = React.useContext(ThemeContext);
   const { language } = React.useContext(LanguageContext);
-  const { setMap } = React.useContext(MapContext);
+  const { setTitle, setColumns, setRows, setLayers, setMetadata } =
+    React.useContext(MapContext);
   const inputRef = React.useRef(null);
 
   const reader = new FileReader();
   reader.addEventListener("loadend", (a) => {
     const result = a.currentTarget.result;
     const decodedResult = decodeURIComponent(result);
     const parsedResult = JSON.parse(JSON.parse(decodedResult));
```

Comment:
I know this is not from this changes but since you're using this code please fix it. Why are you parsing an already pardesd value?